2020-05 depenency updates

Eonfge opened the pull request:
Skipped: 
- pyLast 
- nltk

Both seem to rely on a new build system... Of which I know nothing about. 

@bilelmoussaoui How is your Python knowledge? You might be able to iterate on this.

bilelmoussaoui commented:
Thanks a lot for this! I just looked at pylast & ntk, and both seems to still use setuptools, do you have a specific error message when you bump them to a newer version?

Eonfge commented:
```
Running: python3 setup.py install --prefix=/app
Download error on https://pypi.org/simple/setuptools_scm/: [Errno -3] Temporary failure in name resolution -- Some packages may not be found!
Download error on https://pypi.org/simple/setuptools-scm/: [Errno -3] Temporary failure in name resolution -- Some packages may not be found!
Couldn't find index page for 'setuptools_scm' (maybe misspelled?)
Download error on https://pypi.org/simple/: [Errno -3] Temporary failure in name resolution -- Some packages may not be found!
No local packages or working download links found for setuptools_scm
Traceback (most recent call last):
  File "setup.py", line 44, in <module>
    "Programming Language :: Python :: Implementation :: PyPy",
  File "/usr/lib/python3.7/site-packages/setuptools/__init__.py", line 144, in setup
    _install_setup_requires(attrs)
  File "/usr/lib/python3.7/site-packages/setuptools/__init__.py", line 139, in _install_setup_requires
    dist.fetch_build_eggs(dist.setup_requires)
  File "/usr/lib/python3.7/site-packages/setuptools/dist.py", line 719, in fetch_build_eggs
    replace_conflicting=True,
  File "/usr/lib/python3.7/site-packages/pkg_resources/__init__.py", line 782, in resolve
    replace_conflicting=replace_conflicting
  File "/usr/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1065, in best_match
    return self.obtain(req, installer)
  File "/usr/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1077, in obtain
    return installer(requirement)
  File "/usr/lib/python3.7/site-packages/setuptools/dist.py", line 786, in fetch_build_egg
    return cmd.easy_install(req)
  File "/usr/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 673, in easy_install
    raise DistutilsError(msg)
distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('setuptools_scm')
Error: module pylast: Child process exited with code 1
```

bilelmoussaoui commented:
It seems that one of the two libraries is now requiring a new dependency, setuptools-scm https://pypi.org/project/setuptools-scm/
The easiest solution would be to use flatpak-pip-generator https://github.com/flatpak/flatpak-builder-tools/tree/master/pip to re-generate the pylast & nltk dependencies, which you can copy later to the Lollypop manifest

Eonfge commented:
I've gone the other way. I removed all old dependencies and rebuild them all from scratch. I also moved them to a stand-alone file so all of this becomes easier to maintain in the future.

Now running it locally and it works, but I'm not sure if I have kicked over something critical in the backend.

bilelmoussaoui commented:
> I've gone the other way. I removed all old dependencies and rebuild them all from scratch. I also moved them to a stand-alone file so all of this becomes easier to maintain in the future.
> 
> Now running it locally and it works, but I'm not sure if I have kicked over something critical in the backend.

Sounds like a better solution. As the requirements.txt seems to be updated upstream, I think we are safe to use this and we can update it upstream in case there are any issues :+1:

Eonfge commented:
I'll keep it running for another day. Using some different records and such. If all goes well, I'll merge it tomorrow

Eonfge commented on pypi-dependencies.json:
```diff
@@ -0,0 +1,135 @@
+            ]
+        },
+        {
+            "name": "python3-PyGObject",
```
Could you elaborate? This is based on the upstream requirements.txt

bilelmoussaoui replied:
As you can see, the sources are empty. PyGObject is part of the GNOME runtime already and can't be generated with the flatpak-builder-tools (which results in the empty sources)

Eonfge replied:
Aaah. Done.
